dev/test/index.ts

```diff
@@ -1273,3 +1296,389 @@ describe('getAll() method', () => {
+        });
+        requestCount++;
+        return responsePromise;
+      },
+    };
+
+    return createInstance(overrides);
+  }
+
+  describe('calls getAllDescendants() with correct StructuredQuery', () => {
+    function startAt(name: string): api.IValue {
+      return {
+        referenceValue:
+          DATABASE_ROOT + '/documents/' + name + '/' + REFERENCE_NAME_MIN_ID,
+      };
+    }
+
+    function endAt(name: string): api.IValue {
+      return {
+        referenceValue:
+          DATABASE_ROOT +
+          '/documents/' +
+          name +
+          String.fromCharCode(0) +
+          '/' +
+          REFERENCE_NAME_MIN_ID,
+      };
+    }
+
+    it('for root-level collections', async () => {
+      const overrides: ApiOverride = {
+        runQuery: req => {
+          queryEquals(
+            req,
+            select('__name__'),
+            allDescendants(/* kindless= */ true),
+            fieldFilters(
+              '__name__',
+              'GREATER_THAN_OR_EQUAL',
+              startAt('root'),
+              '__name__',
+              'LESS_THAN',
+              endAt('root')
+            )
+          );
+          return stream();
+        },
+      };
+      firestore = await createInstance(overrides);
+      return firestore.recursiveDelete(firestore.collection('root'));
+    });
+
+    it('for nested collections', async () => {
+      const overrides: ApiOverride = {
+        runQuery: req => {
+          queryEqualsWithParent(
+            req,
+            'root/doc',
+            select('__name__'),
+            allDescendants(/* kindless= */ true),
+            fieldFilters(
+              '__name__',
+              'GREATER_THAN_OR_EQUAL',
+              startAt('root/doc/nestedCol'),
+              '__name__',
+              'LESS_THAN',
+              endAt('root/doc/nestedCol')
+            )
+          );
+          return stream();
+        },
+      };
+      firestore = await createInstance(overrides);
+      return firestore.recursiveDelete(
+        firestore.collection('root/doc/nestedCol')
+      );
+    });
+
+    it('documents', async () => {
+      const overrides: ApiOverride = {
+        runQuery: req => {
+          queryEqualsWithParent(
+            req,
+            'root/doc',
+            select('__name__'),
+            allDescendants(/* kindless= */ true)
+          );
+          return stream();
+        },
+        // Include dummy response for the deleted docRef.
+        batchWrite: () => {
+          return response(successResponse(1));
+        },
+      };
+      firestore = await createInstance(overrides);
+      return firestore.recursiveDelete(firestore.doc('root/doc'));
+    });
+  });
+
+  describe('deletes', () => {
+    it('collection', async () => {
+      firestore = await instantiateInstance([
+        ['anna', 'bob', 'bob/children/charlie', 'bob/children/daniel'],
+      ]);
+      await firestore.recursiveDelete(firestore.collection('collectionId'));
```

Comment:
I am not sure I follow this test. The documents are not part of the collection, but they are still getting returned from the `RunQuery` request if I am not mistaken.